simd/src/Kokkos_SIMD.hpp

```diff
@@ -21,11 +21,17 @@
 
 #include <Kokkos_SIMD_Scalar.hpp>
 
-#ifdef KOKKOS_ARCH_AVX2
+#include <Kokkos_Macros.hpp>
+
+#if defined(KOKKOS_ARCH_AVX) && !defined(__AVX__)
+#error "__AVX__ must be defined for KOKKOS_ARCH_AVX"
+#endif
+#if defined(KOKKOS_ARCH_AVX2) && !defined(__AVX2__)
+#error "__AVX2__ must be defined for KOKKOS_ARCH_AVX2"
 #include <Kokkos_SIMD_AVX2.hpp>
 #endif
```

Comment:
I agree with moving the check into `Kokkos_SIMD.hpp`, but is the condition to include each header correct?

Shouldn't it be something like this?
```suggestion
#if defined(KOKKOS_ARCH_AVX2)
#if !defined(__AVX2__)
#error "__AVX2__ must be defined for KOKKOS_ARCH_AVX2"
#else
#include <Kokkos_SIMD_AVX2.hpp>
#endif
#endif
```

Reply:
Of course, fixed!